### DISCUSSION TITLE
Render different date in each card

### DISCUSSION THREAD

**Discussion by ckskr** [neuefische/questions#122] [Web] [answered]
Hello all,

ich baue eine Diary-App, die verschiedene Inputfelder in Karten auf eine zweite Seite rendert. 
Nun möchte ich gerne bei meinen Diary-Karten **das jeweilige Datum inkl. Uhrzeit rein rendern**. Ich dachte auch es hat schon einmal funktioniert, aber habe mich wohl getäuscht. **Aktuell wird mit jedem neuen Eintrag, die Uhrzeit bei allen Karten auf die neueste geändert.**

Ich verwende **dayjs** und habe es wie folgt integriert. Hier seht ihr den Code aus dem Diary-Eintrag, also dort wo die Einträge gerendert werden. 

Hier noch mein [Vercel](https://capstone-project-pied.vercel.app/diary) & [Github](https://github.com/ckskr/capstone-project)

```
export default function Diary() {
	const deleteEntry = useStore(state => state.deleteEntry);
	const DynamicWrapper = dynamic(() => import('../components/styledWrapper'), {
		ssr: false,
	});
	function Date({datum}) {
		return <StyledDate>{dayjs(datum).format('DD.MM.YYYY h:mm A')}</StyledDate>;
	}
	const entries = useStore(state => state.entries);
	return (
		<DynamicWrapper>
			<DiaryHeadline />
			<StyledDiv>
				{entries &&
					entries.map(entry => {
						return (
							<StyledCard key={entry.id}>
								<Date />
								<StyledFeeling>That is how you felt:</StyledFeeling>
								<ul>
									<StyledLi>{entry.radio1}</StyledLi>
								</ul>
								<StyledH4>You were grateful for:</StyledH4>
								<ul>
									<StyledLi>{entry.first}</StyledLi>
									<StyledLi>{entry.second}</StyledLi>
									<StyledLi>{entry.third}</StyledLi>
								</ul>
								<StyledButton
									variant="delete"
									type="button"
									onClick={() => {
										deleteEntry(entry.id);
									}}
								>
									Delete entry
								</StyledButton>
							</StyledCard>
						);
					})}
			</StyledDiv>
			<Navigation />
		</DynamicWrapper>
	);
}
```

Vielen Dank!

**Comment by mpagels** [comment, marked as answer]
Deine `Date` Komponente erwartet ein `datum` prop, wo wahrscheinlich ein Datum drin sein soll, was angezeigt werden soll. Du übergibst aber kein `prop` beim Aufrufen der `<Date />` Komponente. Daher ist `datum` undefined und `dayjs` ist so nett bei `undefined` einfach das aktuelle, jetzige Datum zu erststellen. Daher wird ständig die aktuelle Zeit angezeigt zu der Zeit, als die Seite gerendert wurde. Probier mal einen Eintrag zu erstellen, warte 1 Minute, gehe auf `Daily entry` und zurück auf `My Diary` und du siehst, wie sich die Zeit ändert. Das Problem liegt also nicht am "überschreiben" wenn ein neuer Eintrag kommt.

Die Lösung ist, dass du generell die Zeit und das Datum des Erstellens speichern musst. Also das Date/Zeit wenn du auf `speichern` drückst. Und dann musst du es halt noch in deine `Date` Komponente übergeben, damit es auch richtig angezeigt wird.

Hoffe, das war verständlich. Wenn du noch Hilfe brauchst, sage gerne Bescheid.

**Reply by ckskr** [reply]
Hi Martin, 

danke dir für die Antwort. 
Wenn ich dich richtig verstanden habe, muss ich also das Datum in der Form integrieren und wie die restlichen Einträge in zustand speichern? :)

**Reply by mpagels** [reply]
Ja genau. In der speicher handle Funktion musst du irgendwie ;-) das Datum abgreifen. Mit dem Date Object oder dayjs. Und das musst du in dein Objekt einbauen, wo du auch die Textinputs speicherst.

**Reply by ckskr** [reply]
An dem "irgendwie" scheitere ich leider... 
Hast du noch einen heißen Tipp für mich? ;)

**Reply by mpagels** [reply]
Da du `dayjs` nutzt, könntest du das nutzen um einen unix timestamp zu erstellen.

```
const timestamp = dayjs().valueOf() // <- gibt zb. 1661849888782 zurück
```

Den Wert speicherst du in deinem Object, wenn du ein Diary Eintrag erstellst. Und mit

```js
const timestamp = 1661849888782
dayjs(timestamp)).format('DD/MM/YYYY') // <- gibt 30/08/2022 zurück
``` 
kannst du den Unix timestamp in ein für Menschen lesbares Datum zurück wandeln.
